mx.truffleruby/mx_truffleruby.py

```diff
@@ -127,7 +127,11 @@ def clean(self, forBuild=False):
             os.remove(self.launcher)
 
     def store_jvm_args(self):
-        distributions = [dep.name for dep in self.subject.buildDependencies]
+        if mx.suite('tools', fatalIfMissing=False):
+            tool_deps = ['tools:CHROMEINSPECTOR', 'tools:TRUFFLE_PROFILER']
+        else:
+            tool_deps = []
+        distributions = [dep.name for dep in self.subject.buildDependencies] + tool_deps
```

Comment:
I think there is a problem with this: I think there no guarantees the tools:CHROMEINSPECTOR and tools:TRUFFLE_PROFILER have been built when we try to get the classpath of them here.
And `mx` will throw if the distribution is not built when trying to get the classpath.
Unless `DEFAULT_DYNAMIC_IMPORTS` forces the entire suite to be built before anything else?
cc @gilles-duboscq

Reply:
They don't seem to be ordered at the top.

Reply:
You can have a look at how [FastR does it for Sulong](https://github.com/oracle/fastr/blob/master/mx.fastr/mx_fastr.py#L585).

I think you could have a same model and dynamically add a `buildDependency` to the projects/distributions that would need this classpath during their build. You can do this during the `mx_post_parse_cmd_line` hook (just define a callable with that name in your `mx_truffleruby` module).

Just like in FastR, be careful not to look for truffleruby projects when truffleruby is a binary suite.

Reply:
I'm not sure this will help @ansalond's use-case, since I expect `tools` is loaded in GraalVM and so we would still have the dependency in the case truffleruby is a source suite.

Reply:
I don't think there's a problem here: if `tools` is loaded the dependency is fine. The important thing is that:
- There's never a `distDependency`
- The `buildDependency` is only there if `tools` is loaded

Reply:
OK. @pitr-ch Could you adopt Gilles' approach?